server/cluster/coordinator.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package cluster
+
+import (
+	"context"
+	"sync"
+
+	"github.com/CeresDB/ceresdbproto/pkg/clusterpb"
+	"github.com/pkg/errors"
+)
+
+type coordinator struct {
```

Comment:
Maybe we need some comments to explain what the coordinator is  and how it works.